### DISCUSSION TITLE
What would be best approach of sending actions to the store.

### DISCUSSION THREAD

**Discussion by vadimkrutovlv** [pointfreeco/swift-composable-architecture#3474] [Q&A]
Hello everyone! 

I recently ran into a situation where I had to update my **Feature** and add some new functionality to it. This functionality simply loads some data from an API **when view appears** and updates the view state based on the response. 

My question is how to better approach this should extend existing **onAppear** action in the reducer or create a new one? Extending existing action allows me to avoid sending an extra message to the **store**, however since I'm almost using only **exhaustive** testing it brakes a lot of existing tests and some of those tests are not even interested in this new functionality because they're testing something else. 

My question is, what is the best way to approach this, should I extend the existing **onAppear** action in the reducer or create a new one? Extending the existing action allows me to avoid sending an extra message to the **store**, however, since I'm pretty much only using **exhaustive** testing, it brakes a lot of existing tests, and some of those tests aren't even interested in this new functionality because they're testing something else.

Creating a new "downloadData" action seems to be reasonable, but it's counterintuitive in the view, because from my understanding actions sent from the view to the store usually should represent real interactions with the view?

Creating a new "downloadData" action seems reasonable, but it feels counterintuitive in the view, since as I understand it, actions sent from the view to the store are usually supposed to represent actual interactions with the view?

**Comment by mbrandonw** [comment]
Hi @vadimkrutovlv, it sounds like you already have an existing `onAppear` action, and so you shouldn't need to add another action. You should simply add this new logic to the existing action. And you definitely should not create actions like `downloadData`. Actions should be named after what happened in the UI rather than what you want the reducer to do.

**Reply by vadimkrutovlv** [reply]
@mbrandonw thank you so much for a quick reply and suggestion I was thinking same way just wasn't sure, what bothers me is that every test which sends onAppear action to the store have to cary a little extra, but I guess if you want exhaustive testing there's nothing you can do...
